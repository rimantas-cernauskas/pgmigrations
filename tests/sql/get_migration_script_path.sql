-- Test function get_migration_script_path()
\i tests/sql/setup/init.sql

BEGIN;
SELECT plan(2);

SET ROLE __pgmigrations_testuser;

SELECT is(
    pgmigrations.get_migration_script_path('migration_dir','up'),
    'migrations/contrib_regression/migration_dir/up.sql',
    'get_migration_script_path() should return relative path the script'
);

PREPARE my_query AS
    SELECT pgmigrations.get_migration_script_path('migration_dir','foobar' );

SELECT throws_ok(
    'my_query',
    'Invalid script type "foobar"',
    'calling get_migration_script_path() with invalid script type should throw exception'
);
DEALLOCATE my_query;

SELECT * FROM finish();
ROLLBACK;