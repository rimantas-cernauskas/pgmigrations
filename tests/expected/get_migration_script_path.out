-- Test function get_migration_script_path()
\i tests/sql/setup/init.sql
\set ECHO errors
1..2
ok 1 - get_migration_script_path() should return relative path the script
ok 2 - calling get_migration_script_path() with invalid script type should throw exception
